app/models/user.rb

```diff
@@ -38,12 +38,39 @@ class User < Common::RedisStore
   # identity attributes, some of these will be overridden by MVI.
   delegate :email, to: :identity, allow_nil: true
   delegate :first_name, to: :identity, allow_nil: true
-  delegate :middle_name, to: :identity, allow_nil: true
-  delegate :last_name, to: :identity, allow_nil: true
-  delegate :gender, to: :identity, allow_nil: true
-  delegate :birth_date, to: :identity, allow_nil: true
-  delegate :zip, to: :identity, allow_nil: true
-  delegate :ssn, to: :identity, allow_nil: true
+
+  def first_name
+    mhv_icn.present? ? mvi&.profile&.given_names&.first : identity.first_name
```

Comment:
Since `mhv_icn` is from `UserIdentity` maybe have an instance of `MVI` in `UserIdentity` and move all the ternary logic there? Or if we want to keep that as a store, have an intermediary class that makes the decision on how to get the data? (Template or Strategy pattern in GoF speak but Ruby may have a more elegant solution).

Reply:
I'm not really crazy about that approach. I prefer to keep UserIdentity distinct from what i'm characterizing as precedence. If anything I would love to move all of these other methods we have on User model and have User be responsible for determining precedence only. For example all of those authorization methods, etc... Also I think Vaffi headers should be in a decorator. No need for it to exist on every single of User that is invoked. If it's needed for EVSS it should fetch user and wrap it in a decorator that adds those methods. User is horribly bloated.

Reply:
Prior to having UserIdentity I was fetching MVI within the SAML User object, we all agreed that was not the right approach. I'm not sure if doing it inside of UserIdentity is much different. From perspective of single responsibility of a class that is.

Reply:
yep agreed on not moving MVI to UserIdentity. I think it's all the repeated `mhv_icn.present?` checks is what could be improved. Could be case for replacing conditionals with polymorphism. Maybe at a later date 😄  for the identity refactor this LGTM 👍